### PR TITLE
Add support for datashader Graph layouts

### PR DESCRIPTION
**Pull request by philippjfr**
The ``layout_nodes`` operation now supports datashader LayoutAlgorithm functions and allows returning a Graph containing the newly computed node positions.

### CODE REVIEW

**Comment by jbednar** [comment]
Looks good to me.

**Comment by philippjfr** [comment]
@jlstevens Ready to merge.

File: holoviews/element/graphs.py [hunk 1]
```diff
@@ -36,22 +42,37 @@ class layout_nodes(Operation):
     """
     Accepts a Graph and lays out the corresponding nodes with the
     supplied networkx layout function. If no layout function is
-    supplied uses a simple circular_layout function.
+    supplied uses a simple circular_layout function. Also supports
+    LayoutAlgorithm function provided in datashader layouts.
     """
 
+    as_graph = param.Boolean(default=False, doc="""
+        Whether to return Nodes or Graph.""")
```
Comment (jlstevens) [inline holoviews/element/graphs.py]: I don't particularly like this name. Why not simply ``graph``?

File: holoviews/element/graphs.py [hunk 2]
```diff
@@ -8,6 +8,12 @@
 from .chart import Points
 from .path import Path
 
+try:
+    from datashader.layout import LayoutAlgorithm as ds_layout
+except:
+    ds_layout = None
+
+
 class graph_redim(redim):
```
Comment (jlstevens) [inline holoviews/element/graphs.py]: Minor point but I would call this ``redim_graph`` instead.

**Comment by jlstevens** [comment]
Looks good other than two simple naming issues I mention in the comments above.

File: holoviews/element/graphs.py (hunk 1 above)
Comment (jlstevens) [reply holoviews/element/graphs.py]: I don't see this parameter used in this PR. Where are you expecting it to be used?

Comment (philippjfr) [reply holoviews/element/graphs.py]: Sure, I'll rename. It's used in ``if self.p.as_graph:`` below.

File: holoviews/element/graphs.py (hunk 2 above)
Comment (philippjfr) [reply holoviews/element/graphs.py]: Not really part of this PR but I'm happy to fix it here if you want me to.

Comment (jlstevens) [reply holoviews/element/graphs.py]: Please do!

File: holoviews/element/graphs.py (hunk 1 above)
Comment (jlstevens) [reply holoviews/element/graphs.py]: I know it is used within this ``redim`` utility. What I'm wondering is where is redim used with this parameter set to ``True`` (assuming it is used with the default value by default).

Comment (philippjfr) [reply holoviews/element/graphs.py]: It's not it's used in the ``layout_nodes`` operation, this PR doesn't touch ``redim`` at all.

Comment (philippjfr) [reply holoviews/element/graphs.py]: Anyway how about ``only_nodes`` or ``nodes_only`` instead of ``as_graph``?

Comment (jlstevens) [reply holoviews/element/graphs.py]: Ah, I got confused and thought it was in ``redim`` for some reason so you can ignore that comment. I prefer ``only_nodes`` though as nodes are always returned how about ``with_edges``?

Comment (philippjfr) [reply holoviews/element/graphs.py]: I've gone with ``only_nodes`` and defaulted to False.

File: holoviews/element/graphs.py (hunk 2 above)
Comment (philippjfr) [reply holoviews/element/graphs.py]: Done.

**Comment by philippjfr** [comment]
Ready if you're happy with it.

**Comment by jbednar** [comment]
Sounds good, thanks!

**Comment by jlstevens** [comment]
Looks good. Happy to merge when the tests pass though I would be even happier if this PR added some new ones!

**Pull request merged**